src/mono/mono/mini/simd-intrinsics.c

```diff
@@ -635,6 +636,18 @@ emit_sri_vector (MonoCompile *cfg, MonoMethod *cmethod, MonoMethodSignature *fsi
 	MonoTypeEnum arg0_type = fsig->param_count > 0 ? get_underlying_type (fsig->params [0]) : MONO_TYPE_VOID;
 
 	switch (id) {
+	case SN_Abs: {
+#ifdef TARGET_ARM64
+        MonoType *arg_type = get_vector_t_elem_type (fsig->params [0]);
+		if (!MONO_TYPE_IS_INTRINSICS_VECTOR_PRIMITIVE (arg_type))
+			return NULL;
+		gboolean is_float = type_is_float (fsig->params [0]);	
```

Comment:
It is faster to make type comparison directly here, since you have `arg0_type` already. If you look at the content of `type_is_float`, that's exactly what it does but with more calls to get `arg0_type`.
```suggestion
		gboolean is_float = arg0_type == MONO_TYPE_R4 || arg0_type == MONO_TYPE_R8;	
```

Reply:
Changed.